### ISSUE TITLE
Response format options

### ISSUE THREAD

**Issue by Sonic2b** [nestjsx/crud#12]
This lib is very cool, but for my project I need to have a certain response format.
Currently, when we request a ressource, we have an array of this ressource as response.
Could we have for the futur some options like the total number of ressources, the number of pages, etc etc...
Because currently can't have a real pagination because we don't have this kind of information. 

Thx.

**Comment by zMotivat0r** [comment]
Hi. Thanks for your interest. The response format will be changed for sure, I just need to implement some convention, I think, so any suggestions appreciated.

**Comment by Sonic2b** [comment]
Maybe, a transformer method between the service and the controller can be useful, with some options...
After the new year I'll try to implement something for my project and I can make a PR if I think it's good

**Comment by zMotivat0r** [comment]
Data transforming is not a problem. The problem is to choose a convention or a specification. [OpenAPI](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md) for example, or any other. Or maybe many of them by implementing some kind of adapters (strategies), so that a developer can choose one and that the response data will be transformed according to the chosen strategy. That's the question.

**Comment by Sonic2b** [comment]
A kind of strategies system sound like the best choice for all spécifications. We can have a default strategy who respect a existant specification and if devs want an other strategy, they have just to create an another.

**Comment by Sonic2b** [comment]
I just see the PR https://github.com/nestjsx/crud/pull/16 
The usage of interceptor can be the a good option, I test the PR on my project and it's very easy to use.

**Comment by zMotivat0r** [comment]
@Sonic2b Yeah, I'm about to merge it, I think.

**Comment by zMotivat0r** [comment]
Ok, I think the best way would be to give an option to the devs, so I'll add a possibility to chose a way of how do they want to receive pagination data: in envelope result body, headers or both. Will to do this during next week maybe.

**Comment by letoche** [comment]
hi,
a good and very flexible way, could be to let the developer provide his own interceptor ?

**Comment by GuihaiHU** [comment]
I customize my response by interceptor as follow.

```
@Injectable()
export class RespInterceptor implements NestInterceptor {
  intercept (
    context: ExecutionContext,
    call$: Observable<any>
  ): Observable<any> {
    return call$.pipe(map(data => {
      if (!(data instanceof Resp) || !(data instanceof Error)) {
        return new Resp(data)
      }
      return data
    }))
  }
}
```
main.ts
```
app.useGlobalInterceptors(new RespInterceptor())
```

**Comment by zMotivat0r** [comment]
Shipped with v3.0.0
